txpipe/diagnostics.py

```diff
@@ -180,21 +189,41 @@ def run(self):
                 plotter.send(None)
             except StopIteration:
                 pass
-
+            
+    def BinEdges(self, data,nbins):
+        import scipy.stats as sts
+        
+        # calculate bin edges for equal number of objects per bin
+        nbquant = np.linspace(0,1,nbins,endpoint=False)
+        edges = sts.mstats.mquantiles(data,prob=nbquant)
+        return edges
+    
+    
     def plot_psf_shear(self):
         # mean shear in bins of PSF
         print("Making PSF shear plot")
         import matplotlib.pyplot as plt
         from scipy import stats
-        from .utils.fitting import fit_straight_line
-
-        delta_gamma = self.config["delta_gamma"]
-        size = 5
-        gr = np.logspace(-3, -2, size // 2)
-        psf_g_edges = np.concatenate([-gr[::-1], gr])
-        print("psf_g_edges", psf_g_edges)
+        import pandas as pd
+        
         psf_prefix = self.config["psf_prefix"]
-
+        #print(psf_prefix)
+        delta_gamma = self.config["delta_gamma"]
+        nbins = self.config["nbins"]
+        g_min = self.config["g_min"]
+        g_max = self.config["g_max"]
+
+        c = self.open_input("shear_catalog")
+        #print(f"{psf_prefix}g1")
+        #pdb.set_trace()
+        #print(f'shear/mcal_psf_g1','AAAAAAAAAAAAAAAAAAAAAAAAAa')
+        #print(c[f"shear/mcal_psf_g1"])
+        #pdb.set_trace()
+        psfg = pd.Series(np.array(c[f"shear/{psf_prefix}g1"]))
```

Comment:
ah yes the reason I have the whole catalog loaded is for the function to calculate the bin edges--it needs all of the data to know where they ought to be to have an equal number of objects in each bin. I may be able to rewrite it such that it can account for the data chunks but that will take me a little bit to figure out